Completion.py

```diff
@@ -107,18 +110,25 @@ def read_settings():
 
     settings["replace_file_path"] = (None, None)
     replace = s.get("ycmd_filepath_replace", {})
+
+    for i,language in enumerate(settings["languages"]):
+        sublime_name = LANG_MAP.get(language)
+        if sublime_name is None:
+            print("[Ycmd] Language `%s` specified in settings file is not supported by ycmd" % language)
+            del settings["languages"][i]
```

Comment:
This wouldn't work - you are removing by index, and so after first ```del``` current loop index would be greater then needed.

I'd better done like this:
1) 
```python
def load_active_languages(settings):
    languages = set(settings["languages"])
    for missing_lang in languages - set(LANG_MAP.keys()):
        print("[Ycmd] [ERROR] Language '%s' specified in settings file is not supported by ycmd" % missing_lang)
        languages.discard(missing_lang)
    return languages
```
2) Use it when filling USER_LANGUAGES like this:
```python
USER_LANGUAGES = load_active_languages(read_settings())
```